contribute/process/commit_template

```diff
@@ -0,0 +1,16 @@
+# <50 character summary of the change
+
+# 1. Why is this change necessary ?
+
+
+# 2. How does this change address the issue ?
+
+# fix: #1234
+# partial-fix: #1234
+# Size/*
+
+# 3. How to verify this change ?
+# Link to some gists, if outputs are long
+
+# 4. What side effects does this change have ?
+# YES / NO issue# created
```

Comment:
add this file to .github/commit_template.md

Reply:
rename as COMMIT_TEMPLATE.md

Reply:
Let us have the sample text for every section. This will help the committer to organise her thoughts better.
The questions should be mandatory along with the answers.

Reply:
@prateekpandey14 I moved it to `.github` directory.
I am not adding any extensions as discussed.